haystack/utils/labels.py

```diff
@@ -75,7 +75,7 @@ def aggregate_labels(
     # Filters define the scope a label is valid for the query, so we group the labels by query and filters.
     grouped_labels: Dict[Tuple, List[Label]] = defaultdict(list)
     for l in labels:
-        label_filter_keys = [f"{k}={''.join(v)}" for k, v in l.filters.items()] if l.filters else []
```

Comment:
The assumption that values of the `Label.filters` dict are always squences was wrong.

Reply:
Tests should be written for this and `aggregate_labels` in general

Reply:
done